src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionValidator.cs

```diff
@@ -49,18 +49,16 @@ public override async Task<SelectionResult> GetValidSelectionAsync(CancellationT
             selectionInfo = CheckErrorCasesAndAppendDescriptions(selectionInfo, root);
 
             // there was a fatal error that we couldn't even do negative preview, return error result
-            if (selectionInfo.Status.FailedWithNoBestEffortSuggestion())
-            {
+            if (selectionInfo.Status.Failed())
```

Comment:
The Failed/FailedWithNoBestEffortSuggestion/Succeeded/SucceededWithSuggestions cases were just hugely complex and difficult to reason about and work with.  The new model is simply: "Succeeded or Failed".  Failed always has messages to show the user.  Succeeded may or may not have messages for the user to know about.